File: app/src/main/java/com/kaiwukj/android/communityhui/di/component/ChatComponent.java
package com.kaiwukj.android.communityhui.di.component;

import com.kaiwukj.android.communityhui.di.module.ChatModule;
import com.kaiwukj.android.communityhui.mvp.contract.ChatContract;
import com.kaiwukj.android.communityhui.mvp.ui.activity.ChatActivity;
import com.kaiwukj.android.communityhui.mvp.ui.fragment.ChatListFragment;
import com.kaiwukj.android.communityhui.mvp.ui.fragment.ChatMessageFragment;
import com.kaiwukj.android.communityhui.mvp.ui.fragment.PushMessageListFragment;
import com.kaiwukj.android.mcas.di.component.AppComponent;
import com.kaiwukj.android.mcas.di.scope.ActivityScope;

import dagger.BindsInstance;
import dagger.Component;


@ActivityScope
@Component(modules = ChatModule.class, dependencies = AppComponent.class)
public interface ChatComponent {
    void inject(ChatActivity activity);
    void inject(ChatListFragment fragment);
    void inject(ChatMessageFragment fragment);
    void inject(PushMessageListFragment fragment);

    @Component.Builder
    interface Builder {
        @BindsInstance
        ChatComponent.Builder view(ChatContract.View view);

        ChatComponent.Builder appComponent(AppComponent appComponent);

        ChatComponent build();
    }
}

File: app/src/main/java/com/kaiwukj/android/communityhui/di/component/ChatComponent.kt
package com.kaiwukj.android.communityhui.di.component

import com.kaiwukj.android.communityhui.di.module.ChatModule
import com.kaiwukj.android.communityhui.mvp.ui.fragment.ChatFragment
import com.kaiwukj.android.mcas.di.component.AppComponent
import com.kaiwukj.android.mcas.di.scope.ActivityScope
import dagger.Component


/**
 * Copyright © KaiWu Technology Company
 * @author Haife
 * @job Android Development
 * @company KW | 开物科技
 * @time 2019/7/16
 * @desc
 */
@ActivityScope
@Component(modules = [ChatModule::class], dependencies = [AppComponent::class])
interface ChatComponent {
    fun inject(activity: ChatFragment)

}
